### PR TITLE
clarified a few notes about 'parallelism'

### PR DESCRIPTION
**Pull request by thoefer**

### CODE REVIEW

File: site/src/main/tut/datatypes/io.md [hunk 1]
```diff
@@ -1195,15 +1195,28 @@ def loop(n: Int): IO[Int] =
 
 ## Parallelism
 
-Since the introduction of the [Parallel](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Parallel.scala) typeclasss in the Cats library and its `IO` instance, it became possible to execute two or more given `IO`s in parallel.
+Since the introduction of the [Parallel](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Parallel.scala) typeclasss in the Cats library and its `IO` instance, it became possible to execute two or more given `IO`s in parallel. "in parallel" here does not refer to any concurrent processing but instead refers to "independent computation" as opposed to monadic computation.
```
Comment (SystemFw) [inline site/src/main/tut/datatypes/io.md]: so, that's not quite the case.
The `Parallel` typeclass is not about parallel execution in general, that's true, but in the case of `IO` (such as in this sentence) "in parallel" does indeed definitely refer to concurrent processing.

Note that `Parallel` itself is not about "independent computation" either (that would be Applicative, in general), but about the link between a monadic type and the applicative instance that can't be consistent with the monadic one.

Comment (thoefer) [reply site/src/main/tut/datatypes/io.md]: Thanks @SystemFw for your input. I'm still wrestling with either the usage of Parallel IO or its documentation - maybe even both ;) I'm confident I understand your explanation and it makes sense to me. I want to update the docs on what we discussed in gitter and here. 

How does one perform concurrent, parallel IO? Testing with the code from a specs test below I expect that my IOs will be executed in parallel in different threads from the provided `ec` using parMapN. As it turns out it is always the same thread. In this case IMO the docs is not correct as it says: for a `ContextShift`it is enought to have an implicit `ExecutionContext` in scope. 

`
import cats.implicits._

implicit val ec = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(20))

val x = IO { println(Thread.currentThread().getName); 15 }

val y = IO { println(Thread.currentThread().getName); 5  }

println("parMapN: " + (x, y).parMapN((a, b) => a + b).unsafeRunSync())

println("mapN: " + (x, y).mapN((a, b) => a + b).unsafeRunSync())

Output:
specs2-4
specs2-4
parMapN: 20
specs2-4
specs2-4
mapN: 20
`

Comment (thoefer) [reply site/src/main/tut/datatypes/io.md]: Looks like one definately needs a context shift, also I think I know what the author of the docs would want to tell us... I'll update my PR.

Comment (SystemFw) [reply site/src/main/tut/datatypes/io.md]: I'll come back to this in more depth later, but:

> As it turns out it is always the same thread. 

It's not a good measure to see if things are concurrent or not. In many cases multiple fibers can run on the same thread.

Comment (thoefer) [reply site/src/main/tut/datatypes/io.md]: You're right, I did not respect fibers in a proper way in my comments. My intention was to find out about how parallel IO works from what is written in the docs. I am looking forward for your feedback!

File: site/src/main/tut/datatypes/io.md [hunk 2]
```diff
@@ -1195,15 +1195,28 @@ def loop(n: Int): IO[Int] =
...
 
+We dispatch our two IO's to a different thread pool and aggregate their results: 
+
+```tut:silent
+import cats.implicits._
+import scala.concurrent.ExecutionContext.Implicits._
+
+val x = IO.shift *> IO(10)
+val y = IO.shift *> IO(5)
+(x, y).parMapN((a, b) => a + b).unsafeRunSync()
```
Comment (alexandru) [inline site/src/main/tut/datatypes/io.md]: You no longer need to `IO.shift` before `parMapN` in 1.0.0.

**Comment by alexandru** [comment]
@thoefer thanks for your contribution, but the two changes are not accurate for the latest version.

Would be happy to take on more docs fixes from you though 🙂 I know we need it.

**Pull request closed without merging**